hydrus/data/resource_based_classes.py

```diff
@@ -0,0 +1,209 @@
+    """
+    database_class = Resource.all_database_classes.get(type_, None)
+    if database_class is None:
+        raise ClassNotFound(type_)
+    return database_class
+
+
+def insert_object(object_: Dict[str, Any], session: scoped_session) -> str:
+    """
+    Insert the object in the database
+    :param object_: Dict containing object properties
+    :param session: sqlalchemy session
+    :return: The ID of the inserted object
+    """
+    type_ = get_type(object_)
+    database_class = get_database_class(type_)
+    id_ = object_.get("id", None)
+    if (
+        id_ is not None and
+        session.query(exists().where(database_class.id == id_)).scalar()
+    ):
+        raise InstanceExists(type_, id_)
+    foreign_keys = database_class.__table__.foreign_keys
+    for fk in foreign_keys:
+        # the name of the column through which this foreign key relationship
+        # is being established
+        fk_column = fk.info["column_name"]
+        try:
+            fk_object = object_[fk_column]
+        except KeyError as e:
+            wrong_property = e.args[0]
+            raise PropertyNotGiven(type_=wrong_property)
+        # insert the foreign key object
+        fk_object_id = insert_object(fk_object, session)
+        # put the id of the foreign instance in this table's column
+        object_[fk_column] = fk_object_id
+    try:
+        # remove the @type from object before using the object to make a
+        # instance of it using sqlalchemy class
+        object_.pop("@type")
+        inserted_object = database_class(**object_)
+    except TypeError as e:
+        # extract the wrong property name from TypeError object
+        wrong_property = e.args[0].split("'")[1]
+        raise PropertyNotFound(type_=wrong_property)
+    try:
+        session.add(inserted_object)
+        session.commit()
+    except Exception as e:
+        # catching any database contraint errors
+        session.rollback()
+        contraint_error = e.orig
+        raise DatabaseConstraintError(contraint_error)
+    return inserted_object.id
+
+
+def get_object(
+    query_info: Dict[str, str], session: scoped_session
+) -> Dict[str, str]:
+    """
+    Get the object from the database
+    :param query_info: Dict containing the id and @type of object that has to retrieved
+    :param session: sqlalchemy session
+    :return: dict of object with its properties
+    """
+    type_ = query_info["@type"]
+    id_ = query_info["id_"]
+    database_class = get_database_class(type_)
+    try:
+        object_ = (
+            session.query(database_class)
+            .filter(database_class.id == id_)
+            .one()
+        ).__dict__
+    except NoResultFound:
+        raise InstanceNotFound(type_=type_, id_=id_)
+    # Remove the unnecessary keys from the object retrieved from database
+    object_template = copy.deepcopy(object_)
+    object_template.pop("_sa_instance_state")
+    object_template.pop("id")
```

Comment:
Why remove ID?

Reply:
Two reasons:
1) The `id` for the object is actually inserted in `crud.py` [here](https://github.com/HTTP-APIs/hydrus/blob/27db73939f2fe8a8d08a8e3c2096876247dcf7d0/hydrus/data/crud.py#L84).
2) In the response we need to have the `id` field as `@id` key, so removing the `id` key.